### PR TITLE
Adds comment at the :name field description about it being the same as `project.app`.

### LINKED ISSUE #506: Improve publishing documentation about `project.package.name` naming constraints
Based on this issue on `Hex` itself: https://github.com/hexpm/hex/issues/364, it turns out that `project.package.name` is supposed to be in snake_case (lowercase with underscores as spaces) because this name is _usually_ taken from the OTP application name, which _usually_ follows this convention.

`Hex` itself seems (I did not check it myself to make sure) to enforce this, but the documentation does not mention it.

Proposal: Alter the information about `project.package.name` on https://hex.pm/docs/publish to read:

> `:name` The name of the package in case you want to publish the package with a different name than the application name. Usually the same as the name of your OTP application (having the same value as `project.app`), written in snake_case (lowercase with underscores between words).

We should probably bicker a little bit about the exact wording of the statement before I'll open a PR that adds it to the page :slightly_smiling_face: .

### PR DESCRIPTION
**Pull request by Qqwy**
Adds comment at the :name field description about it being the same as `project.app`, on the page about publishing your package on Hex.pm.

Fixes #506 .

### CODE REVIEW

File: lib/hexpm/web/templates/docs/publish.html.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ Under the `:package` property are some additional configuration options:
   <dt><code>:licenses</code></dt>
   <dd>A list of licences the project is licensed under. This attribute is required.</dd>
   <dt><code>:name</code></dt>
-  <dd>The name of the package in case you want to publish the package with a different name than the application name.</dd>
+  <dd>The name of the package in case you want to publish the package with a different name than the application name. Usually the same as the name of your OTP application (having the same value as `project.app`), written in snake_case (lowercase with underscores as word separator).</dd>
```
Comment (ericmj) [inline lib/hexpm/web/templates/docs/publish.html.md]: Maybe replace "Usually" with "By default" to make it clear that it's optional?

**Comment by ericmj** [comment]
Ping? :)

**Comment by Qqwy** [comment]
Pong! Thanks for the heads-up.

**Comment by Qqwy** [comment]
All right, it's all there. You can have another glance at this small change :-).

**Pull request merged**